src/Bundle/DependencyInjection/Compiler/RegisterContaoModelPass.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/**
+ * Contao toolkit.
+ *
+ * @package    contao-toolkit
+ * @author     Christopher Bölter <christopher@boelter.eu>
+ * @copyright  2015-2018 netzmacht David Molineus.
+ * @license    LGPL-3.0-or-later https://github.com/netzmacht/contao-toolkit/blob/master/LICENSE
+ * @filesource
+ */
+
+declare(strict_types=1);
+
+namespace Netzmacht\Contao\Toolkit\Bundle\DependencyInjection\Compiler;
+
+use Contao\Model;
+use Netzmacht\Contao\Toolkit\Assertion\Assert;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * Class RegisterContaoModelPass.
+ *
+ * @package Netzmacht\Contao\Toolkit\DependencyInjection\Compiler
+ */
+final class RegisterContaoModelPass implements CompilerPassInterface
+{
+    /**
+     * Name of the tag.
+     *
+     * @var string
+     */
+    private $tagName;
+
+    /**
+     * Index of the argument which should get the tagged references.
+     *
+     * @var int
+     */
+    private $argumentIndex;
+
+    /**
+     * ComponentFactoryCompilePass constructor.
+     *
+     * @param string $tagName       Name of the tag.
+     * @param int    $argumentIndex Index of the argument which should get the tagged references.
+     */
+    public function __construct(string $tagName, int $argumentIndex)
```

Comment:
I don't see any reuse case here, so I would hardcode the tag name and the index in this class.